refactor(theme-default): refactor mediumZoom as <ImgZoom /> in getCustomMDXComponent

Tsukistar opened the pull request:
## Summary
refactor the `plugin-medium-zoom`  to `ImgZoom` in `getCustomMDXComponent` with React 19 Compatibility.
This PR refactors the image component implementation by:
1. Keeping the original component `Img` and add a new component `ImgZoom` to make the image zoomable.
2. Ensuring proper theme integration with CSS variables for background colors.
3. Maintaining full backward compatibility while improving component modularity

The ImgZoom component maintains all previous zoom functionality using medium-zoom library, with enhanced ref forwarding capabilities and proper cleanup logic to prevent memory leaks. Both components are properly exported and registered in the MDX component map for immediate use in documentation.

To add a click-to-zoom image in MDX, use the following:
`<ImgZoom src="./example.png" alt="An image that can be zoomed" />`

## Related Issue

close: https://github.com/web-infra-dev/rspress/issues/2325

<!--- Provide link of related issues -->

## Checklist

<!--- Check and mark with an "x" -->
- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

SoonIter commented on packages/theme-default/src/layout/DocLayout/docComponents/img.tsx:
```diff
@@ -1,6 +1,42 @@
 import { normalizeImagePath } from '@rspress/runtime';
+import mediumZoom from 'medium-zoom';
```
thanks for your contribution, 

what's the difference between `'react-medium-image-zoom'` and `'medium-zoom'`

SoonIter replied:
<img width="400" height="959" alt="image" src="https://github.com/user-attachments/assets/4e3ace6e-bace-4126-8282-467ea21f9fb9" />

`medium-zoom` has not been maintained for a long time. so I recommend `'react-medium-image-zoom'` more

BTW, there is an interesting challenge for you here, which is a bit more complex than this PR, you can continue with it.

You can try using the default value of `mediumZoom` config in `rspress.config.ts` to switch `Img` component  to `ImgZoom`  component by default for users

Here are some tips that you might find useful.

1. you can pass `process.env.MEDIUM_ZOOM` via `initRsbuild.ts` `source.define` from compileTime to runtime

https://github.com/web-infra-dev/rspress/blob/e0b4b135218ad6065c0a1eabf9d7e63a2447f9db/packages/core/src/node/initRsbuild.ts#L226

2. switch `Img` component  to `ImgZoom`  component by default

https://github.com/web-infra-dev/rspress/blob/e0b4b135218ad6065c0a1eabf9d7e63a2447f9db/packages/theme-default/src/layout/DocLayout/docComponents/index.tsx#L11


Some pseudocode

```ts
import { A } from './a';
import { Code } from './code';
import { Hr } from './hr';
import { Img, ImgZoom } from './img';
import { Li, Ol, Ul } from './list';
import { Blockquote, P, Strong } from './paragraph';
import { PreWithCodeButtonGroup } from './pre';
import { Table, Td, Th, Tr } from './table';
import { H1, H2, H3, H4, H5, H6 } from './title';

export function getCustomMDXComponent() {
  return {
    h1: H1,
    h2: H2,
    h3: H3,
    h4: H4,
    h5: H5,
    h6: H6,
    ul: Ul,
    ol: Ol,
    li: Li,
    table: Table,
    td: Td,
    th: Th,
    tr: Tr,
    hr: Hr,
    p: P,
    blockquote: Blockquote,
    strong: Strong,
    a: A,
    code: Code,
    pre: PreWithCodeButtonGroup,
    img: process.env.MEDIUM_ZOOM ? ImgZoom : Img,
  };
}
```

SoonIter commented:
https://github.com/web-infra-dev/rspress/issues/2165

There is an issue that is very similar to this PR, you can also try it out~

Tsukistar commented:
I'll try it after I finish this PR, thank you!

Tsukistar commented:
Hi @SoonIter,

I've completed this PR following your guidance. I've replaced the `medium-zoom` to `react-medium-image-zoom` and I've replaced `options` with `UncontrolledProps` in `react-medium-image-zoom`. This allows users to directly pass any `UncontrolledProps` in `react-medium-image-zoom`, as the way in the `medium-zoom` plugin's documentation (https://github.com/web-infra-dev/rspress/blame/0c109a3b54287ea6b44b04870b229d6953765f4a/packages/document/docs/en/plugin/official-plugins/medium-zoom.mdx#L30-L35C192)

I'm considering whether `process.env.MEDIUM_ZOOM_OPTIONS` is the most suitable method for conveying option information. What are your thoughts on this approach?

SoonIter commented:
Sorry, I made a mistake

There are two image syntaxes in md, after this change, only images with the md syntax `![]()` will take effect.

We also need to use `remark-image` to make the images in <img src=“” /> effective...

```md
<img src="" />

![]()
```

So I might pending this PR, there are some mistakes in my plan.
